texar/torch/modules/encoders/rnn_encoders.py

```diff
@@ -459,15 +459,19 @@ def output_layer(self) -> Optional[nn.Module]:
 
     @property
     def output_size(self) -> int:
+        r"""The feature size of :meth:`forward` output :attr:`outputs`.
+        If output layer does not exist, equals to
+        :attr:`encoder.cell.hidden_size`, otherwise equals to
+        last dimension value of output layer output size.
+        """
         # TODO: We will change the implementation to
         # something that does not require a forward pass.
 
         dim = self._cell.hidden_size
         dummy_tensor = torch.Tensor(dim)
```

Comment:
Let's move the dummy tensor construction inside the `if`-branch. Same for the bi-RNNs.